### DISCUSSION TITLE
How to change markdown texts dynamically like ui.label?

### DISCUSSION THREAD

**Discussion by Anindya088** [zauberzeug/nicegui#2584] [Q&A] [answered]
### Question

Hi All,

I am using both ```ui.label``` and ```ui.markdown``` and recently realized that I can use buttons to update the label text but not the markdown texts. So what will be the correct approach to change markdown texts?

This is the sample code:
```python
label = ui.label('Sample text')
ui.button('Change Text', on_click=lambda: change_text())

def change_text():
    label.text = 'This is new text'
```
This works for ```ui.label``` but if we change this to ```ui.markdown``` then it does not work. Can you please guide me on what will be the correct approach for markdown elements?

Thanks a lot,
Anindya

**Comment by falkoschindler** [comment, marked as answer]
Hi @Anindya088,

The content of a `ui.markdown` element is called `content` instead of `text`:

```py
markdown = ui.markdown('Sample content')
ui.button('Change Content', on_click=lambda: change_content())

def change_content():
    markdown.content = 'This is new content'
```

**Reply by Anindya088** [reply]
Hi @falkoschindler , thanks so much, very clear.

Sorry I got confused because I saw there was a '.text' property for the markdown element, just like the labels. So I thought probably I should use the same one.

Thanks for the quick help on this.
Anindya
